src/formatDate.js

```diff
@@ -9,6 +9,62 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateObj = parseData(date, fromFormat);
+
+  return getData(dateObj, toFormat);
 }
 
+function parseData(date, fromFormat) {
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    const part = date.split(fromFormat[3])[i];
+
+    if (fromFormat[i] === 'DD') {
+      dateObj.day = part;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      dateObj.month = part;
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      dateObj.year = part;
+    }
+
+    if (fromFormat[i] === 'YY') {
+      if (part < 30) {
+        dateObj.year = '20' + part;
+      } else {
+        dateObj.year = '19' + part;
+      }
+    }
+  }
```

Comment:
The current implementation of the parseData function uses multiple if statements which can be inefficient. Consider using a switch statement or an object literal to map format strings to object properties. This will make the code cleaner and potentially more efficient.